core/src/main/java/com/xatkit/core/recognition/nlpjs/mapper/NlpjsIntentMapper.java

```diff
@@ -0,0 +1,107 @@
+import com.xatkit.intent.Context;
+import com.xatkit.intent.ContextParameter;
+import com.xatkit.intent.IntentDefinition;
+import lombok.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static fr.inria.atlanmod.commons.Preconditions.checkNotNull;
+
+public class NlpjsIntentMapper {
+
+    private NlpjsConfiguration configuration;
+
+    private NlpjsEntityReferenceMapper nlpjsEntityReferenceMapper;
+
+    public NlpjsIntentMapper(@NonNull NlpjsConfiguration configuration,
+                             @NonNull NlpjsEntityReferenceMapper nlpjsEntityReferenceMapper) {
+        this.configuration = configuration;
+        this.nlpjsEntityReferenceMapper = nlpjsEntityReferenceMapper;
+
+    }
+
+    public Intent mapIntentDefinition(@NonNull IntentDefinition intentDefinition) {
+        checkNotNull(intentDefinition.getName(), "Cannot map the %s with the provided name %s",
+                IntentDefinition.class.getSimpleName(), intentDefinition.getName());
+        Intent.Builder builder = Intent.newBuilder()
+                .intentName(intentDefinition.getName());
+        List<IntentParameter> intentParameters = new ArrayList<>();
+        List<IntentExample> intentExamples = createIntentExamples(intentDefinition, intentParameters);
+        builder.examples(intentExamples);
+        builder.parameters(intentParameters);
+        return builder.build();
+    }
+
+    private List<IntentExample> createIntentExamples(@NonNull IntentDefinition intentDefinition, @NonNull List<IntentParameter> intentParameters) {
+        List<IntentExample> intentExamples = new ArrayList<>();
+        for (String trainingSentence : intentDefinition.getTrainingSentences()) {
+            intentExamples.add(createIntentExample(trainingSentence, intentDefinition.getOutContexts(),intentParameters));
+        }
+        return intentExamples;
+    }
+
+    private IntentExample createIntentExample(@NonNull String trainingSentence, @NonNull List<Context> outContexts, @NonNull List<IntentParameter> intentParameters) {
+        if (outContexts.isEmpty()) {
+            return new IntentExample(trainingSentence);
+        } else {
+            String preparedTrainingSentence = trainingSentence;
+            for (com.xatkit.intent.Context context : outContexts) {
+                for (ContextParameter parameter : context.getParameters()) {
+                    if (preparedTrainingSentence.contains(parameter.getTextFragment())) {
+                        preparedTrainingSentence = preparedTrainingSentence.replace(parameter.getTextFragment(), "#"
+                                + parameter.getTextFragment() + "#");
+                    }
+                }
+            }
+            String[] splitTrainingSentence = preparedTrainingSentence.split("#");
+            StringBuilder intentExampleBuilder = new StringBuilder();
+            for (int i = 0; i < splitTrainingSentence.length; i++) {
```

Comment:
Use enhanced for here

Reply:
I copied this directly from the DialogFlow implementation.

Reply:
I'll use the enhanced _for_ here. Maybe we should also modify this in the DialogFlow implementation (if you didn't do ytraet)

Reply:
Oops, I'll take care of that!